File: Contracts/Model/Product/Response/PurchaseDetailResponse.cs
﻿namespace Contracts.Model.Product.Response;

public class PurchaseDetailResponse
{
    public int PurchaseNumber { get; set; }
    public string? ProductId { get; set; }
    public int Quantity { get; set; }
    public double UnitPrice { get; set; }
    public string? DiscountId { get; set; } 
    public double TotalPrice => UnitPrice * Quantity;
    public int CartId { get; set; }
}

File: ClientApp/src/app/interfaces/product/userInteract/response/purchaseDetailResponse.ts
import { BaseResponse } from "../../../base-response";
import { PurchaseDetailOptionResponse } from "./purchaseDetailOptionResponse";

export interface PurchaseDetailResponse extends BaseResponse {
    purchaseNumber: number;
    productId: string | null;
    totalOptionsPrice: number;
    purchaseOptions: PurchaseDetailOptionResponse[];
    cartId: number;
    orderId: number | null;
}